src/main/java/com/jcraft/jsch/Session.java

```diff
@@ -624,7 +625,7 @@ private KeyExchange receive_kexinit(Buffer buf) throws Exception {
       throw new JSchException(e.toString(), e);
     }
 
-    kex.doInit(this, V_S, V_C, I_S, I_C);
+    kex.doInit(this, V_S, Util.str2byte(getClientVersion()), I_S, I_C);
```

Comment:
If a user changed the client version string after a Session is connected and a rekeying operation occurred, their SSH connection would then break, because the client version is supposed to be idempotent.

Whilst this could already happen prior, with how you've adapted the Session class to fallback to calling an instance method in the JSch class, we would be making this scenario easier for a user to trigger, since they may naively believe they can change the client version of the JSch instance between multiple calls to `getSession()`.

I'm thinking we should avoid this scenario, by simply initializing the `V_C` property to whatever the value of `jsch.getClientVersion()` is at the time the Session instance is created and avoid trying the callbacks to `jsch.getClientVersion()` within `session.getClientVersion()`.

We simply would need to document to users that the client version of a Session instance is not affected by changes to the JSch instance after the Session is created, and that the only way to manipulate afterwards is by direct calls to `session.setClientVersion()`.

Reply:
Changing the client version while being in an active session was possible before so we've found a possible bug here as far as I can see it. I've solved it now by recreating a member V_C that is filled with the bytes of the client version that is returned by `getClientVersion` at the time of connect.